src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.cs

```diff
@@ -102,47 +102,49 @@ public override void Initialize(AnalysisContext context)
 
         private void OnCompilationStart(CompilationStartAnalysisContext compilationContext)
         {
-            var errors = new List<Diagnostic>();
-            // Switch to "RegisterAdditionalFileAction" available in Microsoft.CodeAnalysis "3.8.x" to report additional file diagnostics: https://github.com/dotnet/roslyn-analyzers/issues/3918
-            if (!TryGetAndValidateApiFiles(compilationContext.Options, compilationContext.Compilation, isPublic: true, compilationContext.CancellationToken, errors, out ApiData publicShippedData, out ApiData publicUnshippedData)
-                || !TryGetAndValidateApiFiles(compilationContext.Options, compilationContext.Compilation, isPublic: false, compilationContext.CancellationToken, errors, out ApiData internalShippedData, out ApiData internalUnshippedData))
+            CheckAndRegisterImplementation(isPublic: true);
+            CheckAndRegisterImplementation(isPublic: false);
```

Comment:
Can a test be added to harness the correct behavior and prevent accidental regressions?

Reply:
Added.